File: src/main/java/br/com/g1bet/mapper/UsuarioMapper.java
package br.com.g1bet.mapper;

import br.com.g1bet.model.Usuario;
import br.com.g1bet.dto.request.UsuarioRequest;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper(componentModel = "spring")
public interface UsuarioMapper {

    UsuarioMapper INSTANCE = Mappers.getMapper(UsuarioMapper.class);

    Usuario toUsuario(UsuarioRequest usuarioRequest);

}

File: src/main/java/br/com/g1bet/mapper/UsuarioMapper.kt
package br.com.g1bet.mapper

import br.com.g1bet.dto.request.UsuarioRequest
import br.com.g1bet.model.Usuario
import org.mapstruct.Mapper
import org.mapstruct.factory.Mappers

@Mapper(componentModel = "spring")
interface UsuarioMapper {
    fun toUsuario(usuarioRequest: UsuarioRequest?): Usuario?

    companion object {
        @JvmField
        val INSTANCE: UsuarioMapper = Mappers.getMapper(UsuarioMapper::class.java)
    }
}